03/Product.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+const Product = ({ data, isDisabled, addItemToCart, removeItemFromCart, name, price, id, isCart, isCategory }) => {
+
+    return (
+        <>
+            {isCategory &&
+                <li key={id}>
+                    <header>{name}</header>
+                    <span>{price}</span>
+                    <button disabled={isDisabled(id)} onClick={() => { addItemToCart(data) }}>Buy</button>
```

Comment:
Może zamiast dodawać obie funkcje tj. `addItemToCart` oraz `removeItemFromCart` to zrobić `prop` o nazwie `action` i tam wstawiać funkcję w zależności od tego czy jest to kategoria czy koszyk.

Zobacz, że wtedy cała struktura powinna się uprościć bo nie będzie powielania całej struktury HTML